include/boost/thread/concurrent_queues/sync_timed_queue.hpp

```diff
@@ -88,8 +88,8 @@ namespace detail
     T pull();
     void pull(T& elem);
 
-    template <class WClock, class Duration>
-    queue_op_status pull_until(chrono::time_point<WClock,Duration> const& tp, T& elem);
+    template <class Duration>
+    queue_op_status pull_until(chrono::time_point<clock,Duration> const& tp, T& elem);
```

Comment:
I'm not saying the implementation was correct, but why you don't want to support other clocks?

Reply:
Currently, if you create a sync_timed_queue<chrono::steady_clock> and then call pull_until() and pass it a chrono::system_clock (or any other clock) time point, the code will fail to compile because it tries to compare `tp` against `clock::now()` in wait_to_pull_until(). So we have two options:

1. Add code to convert between clocks in wait_to_pull_until(), which adds complexity to the code.
2. Change pull_until() to make it clear that mixing clocks is not supported.

I chose number 2 because that's the existing behavior and because it keeps the implementation simpler and because I can't think of a good use-case where someone would need to mix clocks. What do you think?

Reply:
Lets go with approach 2 by now. I will see if I can find out a valid use case.